apps/src/blockly/blockSvgUnused.js

```diff
@@ -0,0 +1,213 @@
+import msg from '@cdo/locale';
+const FRAME_MARGIN_SIDE = 15;
+const FRAME_MARGIN_TOP = 10;
+const FRAME_MARGIN_BOTTOM = 5;
+
+const FRAME_HEADER_HEIGHT = 25;
+
+export const BlockSvgUnused = function(block, helpClickFunc) {
+  this.block_ = block;
+  this.helpClickFunc_ = helpClickFunc;
+
+  this.frameGroup_ = undefined;
+  this.frameClipRect_ = undefined;
+  this.frameBase_ = undefined;
+  this.frameHeader_ = undefined;
+  this.frameText_ = undefined;
+  this.frameHelp_ = undefined;
+
+  this.initChildren();
+};
+
+BlockSvgUnused.UNUSED_BLOCK_HELP_EVENT = 'blocklyUnusedBlockHelpClicked';
+
+BlockSvgUnused.prototype.initChildren = function() {
+  this.frameGroup_ = Blockly.utils.dom.createSvgElement('g', {
+    class: 'blocklyUnusedFrame'
+  });
+
+  var clip = Blockly.utils.dom.createSvgElement(
+    'clipPath',
+    {
+      id: `frameClip${this.block_.id}`
+    },
+    this.frameGroup_
+  );
+  this.frameClipRect_ = Blockly.utils.dom.createSvgElement(
+    'rect',
+    {
+      x: -FRAME_MARGIN_SIDE,
+      y: -(FRAME_MARGIN_TOP + FRAME_HEADER_HEIGHT),
+      height: FRAME_HEADER_HEIGHT
+    },
+    clip
+  );
+
+  this.frameBase_ = Blockly.utils.dom.createSvgElement(
+    'rect',
+    {
+      x: -FRAME_MARGIN_SIDE,
+      y: -(FRAME_MARGIN_TOP + FRAME_HEADER_HEIGHT),
+      fill: '#e7e8ea',
+      stroke: '#c6cacd',
+      rx: 15,
+      ry: 15
+    },
+    this.frameGroup_
+  );
+
+  this.frameHeader_ = Blockly.utils.dom.createSvgElement(
+    'rect',
+    {
+      x: -FRAME_MARGIN_SIDE,
+      y: -(FRAME_MARGIN_TOP + FRAME_HEADER_HEIGHT),
+      fill: '#c6cacd',
+      rx: 15,
+      ry: 15,
+      'clip-path': `url(#frameClip${this.block_.id.replace(/\)/g, '\\)')})`
+    },
+    this.frameGroup_
+  );
+
+  var frameTextVerticalPosition = -(FRAME_MARGIN_TOP + FRAME_HEADER_HEIGHT / 2);
+  if (Blockly.ieVersion()) {
```

Comment:
Is this still a reliable function?  Is there a form of "feature detection" that might be more reliable?